include/BareCpper/ATsamd/DigitalToAnalogConverter_ATsamd5x.hpp

```diff
@@ -0,0 +1,404 @@
+#endif
+
+#include "../Gpio.hpp"
+
+namespace BareCpper
+{
+	namespace SAMD51
+	{
+    /**
+     * DAC voltage reference.
+     * @note Because of silicon bug, the reference
+     * selection for VoltageSupply (VDDANA) is not working.
+     * The voltage supply should be tied to VREF pin and
+     * the reference selection should be UnbufferedExternalReference.
+     * 
+     */
+    enum class DacReference : uint32_t
+    {
+      UnbufferedExternalReference,
+      VoltageSupply,
+      BufferedExternalReference,
+      Internal1V0                 = SUPC_VREF_SEL_1V0_Val,
+			Internal1V1                 = SUPC_VREF_SEL_1V1_Val,
+			Internal1V2                 = SUPC_VREF_SEL_1V2_Val,
+			Internal1V25                = SUPC_VREF_SEL_1V25_Val,
+			Internal2V0                 = SUPC_VREF_SEL_2V0_Val,
+			Internal2V2                 = SUPC_VREF_SEL_2V2_Val,
+			Internal2V4                 = SUPC_VREF_SEL_2V4_Val,
+			Internal2V5                 = SUPC_VREF_SEL_2V5_Val,
+    };
+
+    /**
+     * Possible oversampling DAC ratios.
+     * Multiplies the sample rate.
+     * 
+     */
+    enum class DacOversamplingRatio : uint8_t
+    {
+      Osr1  = 0x0,
+      Osr2  = 0x1,
+      Osr4  = 0x2,
+      Osr8  = 0x3,
+      Osr16 = 0x4,
+      Osr32 = 0x5
+    };
+
+    /**
+     * Set output buffer current based on
+     * DAC clock.
+     * 
+     */
+    enum class DacCurrentControl : uint8_t
+    {
+      Cc100k  = 0x0,
+      Cc1M    = 0x1,
+      Cc12M   = 0x2
+    };
+
+    /**
+     * DAC controller interface.
+     * Controls the two DACs inside the MCU.
+     * 
+     */
+    class DigitalToAnalogConverter
+    {
+      public:
+        static constexpr uint16_t maxValue = 4095;  //< 12-bit resolution
+
+        /**
+         * Initialize the DAC by selecting the clock source
+         * and setting the correct current control values and voltage reference
+         */
+        void init()
+        {
+          // enable clocks for DAC
+          MCLK->APBDMASK.reg |= MCLK_APBDMASK_DAC;
+          // ensure GCLK4 is configured for 12 MHz clock output
+					GCLK->PCHCTRL[DAC_GCLK_ID].reg = GCLK_PCHCTRL_GEN_GCLK4_Val | (1 << GCLK_PCHCTRL_CHEN_Pos);
+          // Wait for clock synchronization
+				  while (GCLK->SYNCBUSY.reg);
+          // reset the peripheral
+          reset();
+          // setup current control
+          DAC->DACCTRL[0].reg = static_cast<uint8_t>(DacCurrentControl::Cc12M) << DAC_DACCTRL_CCTRL_Pos;
+          DAC->DACCTRL[1].reg = static_cast<uint8_t>(DacCurrentControl::Cc12M) << DAC_DACCTRL_CCTRL_Pos;
+        }
+
+        /**
+         * Enable the DAC controller and wait for the
+         * startup time of the DACs to pass.
+         * 
+         */
+        void enable()
+        {
+          DAC->CTRLA.bit.ENABLE = 1;
+          while(DAC->SYNCBUSY.bit.ENABLE);
+          // wait for the startup time to finish for both DACs
+          ///@note due to silicon bug, check INTFLAG EMPTY 0 and EMPTY 1
```

Comment:
Comment: is there a bug reference for the Errata?

Reply:
Added comment (errata 2.9.6).